index.js

```diff
@@ -83,6 +84,8 @@ const jsConditionalRedirect = (jsRedirect, noscriptRedirect) => {
 }
 
 const handler = async (event, context, callback) => {
+  await loadEnvVars()
```

Comment:
This is doing two serial calls to kms decrypt, so we really shouldn't call it on each event; It should be called once on initialization and never again. Note that, because you're currently overwriting the orig env keys, you'll have to be careful not to re-decrypt something you've already decrypted. Elsewhere we manage this by prefixing encrypted env keys with `ENCRYPTED_` and placing the decrypted value in a new key without that prefix.

Reply:
I've changed the strategy to only load env vars once and only decrypt as needed, so I think this is fixed